### PR TITLE
Comment func remaining

### PR DESCRIPTION
**Pull request by Developers-codz**
This PR includes single post page, comment delete and edit functionality and also filter functionality 
Active Navlink and some small fixes as well

### CODE REVIEW

File: src/Redux/Reducers/commentSlice.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+
+const initialState = {
+  comments:[],
+  editModalOpen: false
+};
+export const getComments = createAsyncThunk("/comments/getComments",async (id,{rejectWithValue})=>{
+  try{
+    const response =  await axios.get(`/api/comments/${id}`)
+    console.log(response.data)
```
Comment (maitrakhatri) [inline src/Redux/Reducers/commentSlice.js]: remove unnecessary console logs

File: src/Redux/Reducers/postsSlice.js [hunk 2]
```diff
@@ -21,6 +22,17 @@ export const getAllPosts = createAsyncThunk(
   }
 );
 
+export const getSinglePostData = createAsyncThunk("posts/getSinglePostData", async (id,{rejectWithValue}) =>{
+  try{
+    const response = await axios.get(`/api/posts/${id}`);
+    console.log(response.data)
```
Comment (maitrakhatri) [inline src/Redux/Reducers/postsSlice.js]: same as above

File: src/Redux/Reducers/postsSlice.js [hunk 3]
```diff
@@ -97,6 +109,46 @@ export const commentPost = createAsyncThunk(
   }
 );
 
+export const deletePostComment = createAsyncThunk(
+  "posts/deletePostComment",
+  async (details) => {
+    const {postId, commentId} = details;
+    console.log( "post " + postId , "comment "+ commentId)
```
Comment (maitrakhatri) [inline src/Redux/Reducers/postsSlice.js]: same as above

File: src/Redux/Reducers/postsSlice.js [hunk 4]
```diff
@@ -97,6 +109,46 @@ export const commentPost = createAsyncThunk(
...
+    console.log(details)
+    const {postId, commentId,commentData} = details;
+    const encodedToken = localStorage.getItem("token");
+    try {
+      const response = await axios.post(
+        `/api/comments/edit/${postId}/${commentId}`,
+        { content: commentData },
+        { headers: { authorization: encodedToken } }
+      );
+      console.log(response.data)
```
Comment (maitrakhatri) [inline src/Redux/Reducers/postsSlice.js]: same

File: src/Redux/Reducers/postsSlice.js [hunk 5]
```diff
@@ -177,12 +230,14 @@ const postsSlice = createSlice({
...
     getPostToEdit: (state, action) => {
       state.postToEdit = action.payload;
-      console.log(state.postToEdit);
     },
     setEditEmpty: (state, action) => {
       state.postToEdit = null;
-      console.log(state.postToEdit);
     },
+    setSortBy:(state,action) =>{
+      console.log(action.payload)
```
Comment (maitrakhatri) [inline src/Redux/Reducers/postsSlice.js]: same

File: src/components/modal/EditCommentModal.jsx [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+import { editPostComment } from "../../Redux/Reducers/postsSlice";
+import {toggleEditModal} from "../../Redux/Reducers/commentSlice"
+
+export const EditCommentModal = ({
+  postId,
+  comment,
+}) => {
+  const { theme } = useTheme();
+  const {editModalOpen} = useSelector((store) => store.comments)
+  console.log(editModalOpen)
```
Comment (maitrakhatri) [inline src/components/modal/EditCommentModal.jsx]: here too

File: src/components/modal/Modal.jsx [hunk 7]
```diff
@@ -20,6 +22,15 @@ export const Modal = ({ isModalOpen, setModalOpen }) => {
   const postToEdit = useSelector((store) => store?.posts?.postToEdit);
   const [postData, setPostData] = useState({ content: "" });
   const [editData, setEditData] = useState(null);
+  const [showPicker, setShowPicker] = useState(false);
+
+  const onEmojiClick = (event, emojiObject) => {
+    const newData = postData.content + emojiObject.emoji;
+    console.log(emojiObject);
```
Comment (maitrakhatri) [inline src/components/modal/Modal.jsx]: here

File: src/Assets/icons/Logo.jsx [hunk 8]
```diff
@@ -2,7 +2,7 @@ import { useTheme } from "../../context/theme-context"
 export const Logo = ({width,height})=>{
     const {theme} = useTheme();
     return(
-        <svg width={width} height={height} viewBox="0 0 512 512" color={theme==="dark" ? "#d6806a":"#ca2535"}>
+        <svg width={width} height={height} viewBox="0 0 512 512" color="#d6806a">
```
Comment (Kothalkarpriya) [inline src/Assets/icons/Logo.jsx:5]: Inline styling is not a good idea to go with. Try accessing styling using class

**Review by maitrakhatri — COMMENTED** [review]
Edit functionality is not working, edit input field should pre filled with the comment, it is coming empty.
Filter/Sort posts by latest, oldest, trending is also not working in the preview.

Remove all the unnecessary console logs, almost all API calls and Slices are console logging the response.

Rest looks good

File: src/Redux/Reducers/postsSlice.js [hunk 9]
```diff
@@ -97,6 +109,46 @@ export const commentPost = createAsyncThunk(
   }
 );
 
+export const deletePostComment = createAsyncThunk(
+  "posts/deletePostComment",
+  async (details) => {
```
Comment (Kothalkarpriya) [inline src/Redux/Reducers/postsSlice.js:113]: Good that you are using async await

**Review by Kothalkarpriya — APPROVED** [review]
Try to create small PR. COde is clean and good , you can merge

File: src/Redux/Reducers/postsSlice.js (hunk 9 above)
Comment (Developers-codz) [reply src/Redux/Reducers/postsSlice.js:113]: Thank you

File: src/Assets/icons/Logo.jsx (hunk 8 above)
Comment (Developers-codz) [reply src/Assets/icons/Logo.jsx:5]: For icons I am taking height and width props and then setting the height width attributes accordingly

**Comment by Developers-codz** [comment]
> Edit functionality is not working, edit input field should pre filled with the comment, it is coming empty. Filter/Sort posts by latest, oldest, trending is also not working in the preview.
> 
> Remove all the unnecessary console logs, almost all API calls and Slices are console logging the response.
> 
> Rest looks good

Let me check the filters it was working properly in my local machine, Its working now you can check and approve

**Pull request merged**